Code/GraphMol/Canon.cpp

```diff
@@ -153,75 +150,70 @@ void canonicalizeDoubleBond(Bond *dblBond, INT_VECT &bondVisitOrders,
                    atomVisitOrders[dblBond->getEndAtomIdx()] > 0,
                "neither end atom traversed");
 
-  // atom1 is the lower numbered atom of the double bond (the one traversed
-  // first)
-  Atom *atom1, *atom2;
-  if (atomVisitOrders[dblBond->getBeginAtomIdx()] <
-      atomVisitOrders[dblBond->getEndAtomIdx()]) {
-    atom1 = dblBond->getBeginAtom();
-    atom2 = dblBond->getEndAtom();
-  } else {
-    atom1 = dblBond->getEndAtom();
-    atom2 = dblBond->getBeginAtom();
-  }
-
+  Atom *atom1 = dblBond->getBeginAtom();
+  Atom *atom2 = dblBond->getEndAtom();
   // we only worry about double bonds that begin and end at atoms
   // of degree 2 or 3:
   if ((atom1->getDegree() != 2 && atom1->getDegree() != 3) ||
       (atom2->getDegree() != 2 && atom2->getDegree() != 3)) {
     return;
   }
+  // ensure that atom1 is the lower numbered atom of the double bond (the one
+  // traversed first)
+  if (atomVisitOrders[dblBond->getBeginAtomIdx()] >=
+      atomVisitOrders[dblBond->getEndAtomIdx()]) {
+    std::swap(atom1, atom2);
+  }
 
   Bond *firstFromAtom1 = nullptr, *secondFromAtom1 = nullptr;
   Bond *firstFromAtom2 = nullptr, *secondFromAtom2 = nullptr;
 
-  int firstVisitOrder = 100000;
-
   ROMol &mol = dblBond->getOwningMol();
+  int firstVisitOrder = mol.getNumBonds() + 1;
```

Comment:
This should probably be unsigned int

Reply:
fixing this, along with a number of other signed/unsigned things